src/conduits/java/com/enderio/conduits/common/integrations/ae2/AE2InWorldConduitNodeHost.java

```diff
@@ -0,0 +1,93 @@
+package com.enderio.conduits.common.integrations.ae2;
+
+import appeng.api.config.SecurityPermissions;
+import appeng.api.networking.*;
+import appeng.api.util.AECableType;
+import com.enderio.EnderIO;
+import com.enderio.api.conduit.IConduitType;
+import com.enderio.api.conduit.IExtendedConduitData;
+import com.enderio.conduits.common.blockentity.ConduitBlockEntity;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+public class AE2InWorldConduitNodeHost implements IInWorldGridNodeHost, IExtendedConduitData<AE2InWorldConduitNodeHost> {
+
+    private final Level level;
+    private final BlockPos pos;
+    private final AE2ConduitType type;
+    private final IManagedGridNode mainNode;
+
+    public AE2InWorldConduitNodeHost(Level level, BlockPos pos, AE2ConduitType type) {
+        this.level = level;
+        this.pos = pos;
+        this.type = type;
+        mainNode = GridHelper.createManagedNode(this, new GridNodeListener())
+            .setVisualRepresentation(type.getConduitItem())
+            .setInWorldNode(true)
+            .setTagName("conduit");
+        if (type.isDense()) {
+            mainNode.setFlags(GridFlags.DENSE_CAPACITY);
+        }
+    }
+
+    @Nullable
+    @Override
+    public IGridNode getGridNode(Direction dir) {
+        return mainNode.getNode();
+    }
+
+    @Override
+    public AECableType getCableConnectionType(Direction dir) {
+        if (type.isDense())
+            return AECableType.DENSE_SMART;
+        return AECableType.SMART;
+    }
+
+    @Override
+    public CompoundTag serializeNBT() {
+        CompoundTag nbt = new CompoundTag();
+        mainNode.saveToNBT(nbt);
+        return nbt;
+    }
+
+    @Override
+    public void deserializeNBT(CompoundTag nbt) {
+        mainNode.loadFromNBT(nbt);
+    }
+
+    void onSecurityBreach() {
```

Comment:
Note for 1.20.1 porting: This will all be binned I believe as AE2 doesnt have security stuff anymore.